### PR TITLE
Updating docs

### PR DESCRIPTION
**Pull request by seanmalloy-crest**
Updating the toctrees so that we can maintain the same directory structure we follow in SMartX in API reference.

### CODE REVIEW

File: docs/components/index.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+base_components
```
Comment (jayjoshi-crest) [inline docs/components/index.rst:1]: Please Name it Components instead of base_components

**Review by jayjoshi-crest — APPROVED** [review]

File: docs/components/index.rst (hunk 1 above)
Comment (seanmalloy-crest) [reply docs/components/index.rst]: Change made here: 90e4dd4

**Review by bhargav-nariyani-crest — APPROVED** [review]

**Pull request merged**